components/Container/index.tsx

```diff
@@ -0,0 +1,292 @@
+'use client'
+
+import React, {
+  CSSProperties,
+  useState,
+  useEffect,
+  useRef,
+} from 'react';
+
+import {
+  Button,
+} from 'react-aria-components';
+
+import { DmpIcon } from '@/components/Icons';
+import {
+  useResponsive,
+  getSizeByName,
+} from '@/hooks/responsive';
+
+import './containers.scss';
+
+
+type DirectionType =
+  | null
+  | "left"
+  | "right";
+
+// Extend the CSSProps so that we can actually add css variables to tags.
+type CustomCSSProperties = CSSProperties & Record<string, string>;
+
+/**
+ * This is the base layout component. Our custom layout components should
+ * inherrit from this one. So that we have some common functionality for all
+ * layout containers.
+ */
+interface LayoutContainerProps extends React.HTMLAttributes<HTMLDivElement> {
+  style?: CustomCSSProperties,
+}
+
+export const LayoutContainer: React.FC<LayoutContainerProps> = ({
+  id,
+  className,
+  style,
+  children,
+}) => {
+  return (
+    <>
+      <div
+        id={id}
+        className={`layout-container ${className}`}
+        style={style}
+      >
+        {children}
+      </div>
+    </>
+  )
+}
+
+
+/**
+ * ContentContainer is the base component that is meant to hold content
+ * inside a LayoutContainer.
+ */
+interface ContentContainerProps extends React.HTMLAttributes<HTMLDivElement> {
+  style?: CustomCSSProperties,
+}
+
+export const ContentContainer: React.FC<ContentContainerProps> = ({
+  children,
+  id = '',
+  className = '',
+  style,
+}) => {
+  return (
+    <>
+      <div
+        id={id}
+        className={`layout-content-container ${className}`}
+        style={style}
+      >
+        {children}
+      </div>
+    </>
+  )
+}
+
+
+/**
+ * A toolbar that will reside inside a layout container.
+ */
+export const ToolbarContainer: React.FC<LayoutContainerProps> = ({
+  children,
+  style,
+  id = '',
+  className = '',
+}) => {
+  return (
+    <>
+      <div
+        id={id}
+        className={`layout-container layout-toolbar-container ${className}`}
+        style={style}
+      >
+        {children}
+      </div>
+    </>
+  )
+}
+
+
+/**
+ * Layout with a dynamic sidebar
+ */
+interface LayoutWithSidebarProps extends LayoutContainerProps {
+}
+
+export const LayoutWithSidebar: React.FC<LayoutWithSidebarProps> = ({
+  children,
+  id,
+  className = "",
+}) => {
+  const thisRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (thisRef) {
+      if (thisRef.current) {
+        let direction: DirectionType = null;
+
+        children?.forEach((rNode) => {
```

Comment:
"children" prop is not always an array. You should probably add a check before using "forEach". Also, use a type guard check that "rNode" is a valid React element:
`	if (React.isValidElement(rNode) && rNode.type && typeof rNode.type === 'function') { const { name } = rNode.type as React.FunctionComponent;`